tests/cloud_answer_tests_runner.py

```diff
@@ -0,0 +1,390 @@
+        eg. {'Actual': '/usr/tmp/tmp43la9b0w.png',
+             'Expected': '/usr/tmp/tmpbpaqbgi3.png',
+             'Difference': '/usr/tmp/tmp43la9b0w-failed-diff.png'}
+    None
+        When `error_string` does not conform to yt's answer-testing error
+        message, which has the information for plot file locations on disk.
+
+    """
+    unknown_failure = False
+    base_regex = r"\s*\n\s*(.*?.png)"
+    img_regex = {"Actual": "Actual:" + base_regex,
+                 "Expected": "Expected:" + base_regex,
+                 "Difference": "Difference:" + base_regex}
+    img_path = {}
+    for key in img_regex:
+        result = re.search(img_regex[key], error_string, re.MULTILINE)
+        if not result:
+            unknown_failure = True
+            break
+        # store the locations of actual, expected and diff plot files
+        img_path[key] = result.group(1)
+
+    if not unknown_failure:
+        return img_path
+    return None
+
+def parse_nose_xml(nose_xml):
+    """Parse xml file generated by nosetests.
+
+    Parse nose xml file to find following details:
+     Failed tests: These could be due to difference in golden answer image and
+                   corresponding test plot.
+     Missing tests: These errors occur when a corresponding golden answer image
+                    is not found.
+
+    Parameters
+    ----------
+    nose_xml : string
+        full path of xml file to be parsed
+
+    Returns
+    -------
+    tuple : (failed_answers, missing_answers)
+
+        failed_answers : list of tuples (string, dict)
+        Collection of tuples where the first part is a string denoting the
+        test name, the second part is a dictionary that stores the actual,
+        expected and difference plot file locations of the test.
+        eg. [('yt.visualization.tests.test_line_plots:test_line_plot',
+                {'Actual': '/usr/tmp/tmp43la9b0w.png',
+                'Expected': '/usr/tmp/tmpbpaqbgi3.png',
+                'Difference': '/usr/tmp/tmp43la9b0w-failed-diff.png'}
+            )]
+
+        missing_answers : list of string
+        Collection of missing answer tests specifying full name of the test.
+        eg. ['yt.visualization.tests.test_line_plots:test_multi_line_plot']
+
+    """
+    missing_answers = []
+    failed_answers = []
+
+    tree = ET.parse(nose_xml)
+    testsuite = tree.getroot()
+
+    for testcase in testsuite:
+        for error in testcase:
+            test_name = testcase.attrib["classname"] + ":" \
+                        + testcase.attrib["name"]
+            if "No such file or directory" in error.attrib["message"]:
+                missing_answers.append(test_name)
+            elif "Items are not equal" in error.attrib["message"]:
+                img_path = extract_image_locations(error.attrib["message"])
+                if img_path:
+                    failed_answers.append((test_name, img_path))
+    return failed_answers, missing_answers
```

Comment:
This whole thing is pretty unobjectionable, but I have two questions:

 - Is it possible to put actual boolean metadata into the test xml output via our plugin, so that we don't have to scan the error message?
 - Does nose provide any utilities for parsing its own output?